viadot/sources/mindful.py

```diff
@@ -0,0 +1,251 @@
+            self.logger.warning(
+                f"Mindful 'start_date' and 'end_date' parameter are None or not in datetime format, redefined to:\n\t\t{start_date}\n\t\t{end_date}"
+            )
+        elif isinstance(start_date, datetime) and not isinstance(end_date, datetime):
+            end_date = start_date + timedelta(days=date_interval)
+            if end_date > datetime.now():
+                end_date = datetime.now()
+            self.logger.info(
+                f"Mindful 'end_date' variable is None or not in datetime format, it has been taken as: {self.end_date}."
+            )
+        elif start_date >= end_date:
+            raise ValueError(
+                "'start_date' parameter must be lower than 'end_date' variable."
+            )
+        else:
+            self.logger.info(
+                f"Mindful files to download will store data from {start_date} to {end_date}."
+            )
+
+        return start_date, end_date
+
+    def _mindful_api_response(
+        self,
+        params: Dict[str, Any] = None,
+        endpoint: str = "",
+        **kwargs,
+    ) -> Response:
+        """Basic call to Mindful API given an endpoint.
+
+        Args:
+            params (Dict[str, Any], optional): Parameters to be passed into the request. Defaults to None.
+            endpoint (str, optional): API endpoint for an individual request. Defaults to "".
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+
+        response = handle_api_response(
+            url=f"https://{self.region}surveydynamix.com/api/{endpoint}",
+            params=params,
+            headers=self.header,
+            method="GET",
+        )
+
+        return response
+
+    def get_response_df(
+        self,
+        endpoint: str = "",
+        start_date: datetime = None,
+        end_date: datetime = None,
+        date_interval: int = 1,
+        limit: int = 1000,
+        **kwargs,
+    ) -> Response:
+        """Gets a list of survey interactions as a JSON array of interaction resources.
+
+        Args:
+            endpoint (str, optional): API endpoint for an individual request. Defaults to ".
+            start_date (datetime, optional): Start date of the request. Defaults to None.
+            end_date (datetime, optional): End date of the resquest. Defaults to None.
+            date_interval (int, optional): How many days are included in the request.
+                If end_date is passed as an argument, date_interval will be invalidated. Defaults to 1.
+            limit (int, optional): The number of matching interactions to return. Defaults to 1000.
+
+        Raises:
+            ValueError: Not available endpoint.
+            APIError: Failed to download data from endpoint.
+
+        Returns:
+            Response: request object with the response from the Mindful API.
+        """
+        if endpoint not in self.ENDPOINTS:
```

Comment:
Since its a class attribure, it's simpler to refer to it with `Mindful.ENDPOINTS`.